advocacy_docs/supported-open-source/pgbackrest/08-multiple-repositories.mdx

```diff
@@ -0,0 +1,590 @@
+/var/lib/edb/as13/backups/repo1/archive/demo/13-1/00000002.history
+/var/lib/edb/as13/backups/repo2/archive/demo/13-1/00000002.history
+```
+
+PostgreSQL found the WALs needed for recovery in our **repo2**, picked a new timeline and pushed the history file to the repositories.
+
+Every time a new timeline is created, PostgreSQL creates a timeline history file that shows which timeline it branched off from and when. These history files are necessary to allow the system to pick the right WAL segment files when recovering from a backup that contains multiple timelines.
+
+```bash
+root# systemctl stop edb-as-13.service
+$ rm -rf /var/lib/edb/as13/data/*
+$ rm -rf /var/lib/edb/as13/backups/repo1/archive/demo/13-1/00000002.history
+$ pgbackrest restore --stanza=demo --target="2021-04-19 14:38:55.313716+00:00" --type=time --target-timeline=current --no-delta --target-action=promote
+P00   INFO: restore command begin ...
+P00   INFO: repo2: restore backup set 20210419-143749F
+P00   INFO: write updated /var/lib/edb/as13/data/postgresql.auto.conf
+P00   INFO: restore global/pg_control (performed last to ensure aborted restores cannot be started)
+P00   INFO: restore command end: completed successfully
+
+root# systemctl start edb-as-13.service
+$ cat /var/lib/edb/as13/data/log/edb-2021-04-19_145721.log
+P00   INFO: found 00000002.history in the repo2: 13-1 archive
+P00   INFO: unable to find 00000003.history in the archive
+LOG:  selected new timeline ID: 3
+P00   INFO: pushed WAL file '00000003.history' to the archive
+```
+
+As shown above, since the history file is stored within the same repository as the restored backup set, PostgreSQL is able to pick a new and accurate timeline.
+
+Let's now retry after moving the history file to **repo1**.
+
+```bash
+root# systemctl stop edb-as-13.service
+$ rm -rf /var/lib/edb/as13/data/*
+$ mv /var/lib/edb/as13/backups/repo2/archive/demo/13-1/00000002.history /var/lib/edb/as13/backups/repo1/archive/demo/13-1/00000002.history
+$ pgbackrest restore --stanza=demo --target="2021-04-19 14:38:55.313716+00:00" --type=time --target-timeline=current --no-delta --target-action=promote
+root# systemctl start edb-as-13.service
+$ cat /var/lib/edb/as13/data/log/edb-2021-04-19_150037.log
+P00   INFO: found 00000002.history in the repo1: 13-1 archive
+P00   INFO: found 00000003.history in the repo1: 13-1 archive
+P00   INFO: unable to find 00000004.history in the archive
+LOG:  selected new timeline ID: 4
+P00   INFO: pushed WAL file '00000004.history' to the archive
+```
+
+As we can see, both `00000002.history` and `00000003.history` have been found in **repo1** so PostgreSQL could pick the next timeline correctly, even if the restored backup set came from **repo2**.
+
+The `archive-get` enhancement will then tolerate gaps in one repository (due to lack of disk space e.g.) mainly because it will still be able to find the missing files in the other repository.
```

Comment:
"then tolerate gaps" please re-phrase

Reply:
@ManjushaVaidya: "When using multiple repositories, the `archive-get` command will tolerate gaps..."  ?